src/wp-includes/class-wp-duotone.php

```diff
@@ -948,45 +944,41 @@ public static function register_duotone_support( $block_type ) {
 	 * This handles selectors defined in `color.__experimentalDuotone` support
 	 * if `filter.duotone` support is not defined.
 	 *
-	 * @param string $block_name The block name.
+	 * @param WP_Block_Type $block_type Block type to check for support.
 	 *
 	 * @internal
 	 *
 	 * @since 6.3.0
 	 *
 	 * @return string The CSS selector or null if there is no support.
 	 */
-	private static function get_selector( $block_name ) {
-		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
-
-		if ( $block_type && property_exists( $block_type, 'supports' ) ) {
-			/*
-			 * Backwards compatibility with `supports.color.__experimentalDuotone`
-			 * is provided via the `block_type_metadata_settings` filter. If
-			 * `supports.filter.duotone` has not been set and the experimental
-			 * property has been, the experimental property value is copied into
-			 * `supports.filter.duotone`.
-			 */
-			$duotone_support = _wp_array_get( $block_type->supports, array( 'filter', 'duotone' ), false );
-			if ( ! $duotone_support ) {
-				return null;
-			}
-
-			/*
-			 * If the experimental duotone support was set, that value is to be
-			 * treated as a selector and requires scoping.
-			 */
-			$experimental_duotone = _wp_array_get( $block_type->supports, array( 'color', '__experimentalDuotone' ), false );
-			if ( $experimental_duotone ) {
-				$root_selector = wp_get_block_css_selector( $block_type );
-				return is_string( $experimental_duotone )
-					? WP_Theme_JSON::scope_selector( $root_selector, $experimental_duotone )
-					: $root_selector;
-			}
+	private static function get_selector( $block_type ) {
+		/*
+		 * Backwards compatibility with `supports.color.__experimentalDuotone`
+		 * is provided via the `block_type_metadata_settings` filter. If
+		 * `supports.filter.duotone` has not been set and the experimental
+		 * property has been, the experimental property value is copied into
+		 * `supports.filter.duotone`.
+		 */
+		$duotone_support = block_has_support( $block_type, array( 'filter', 'duotone' ) );
+		if ( ! $duotone_support ) {
+			return null;
+		}
 
-			// Regular filter.duotone support uses filter.duotone selectors with fallbacks.
-			return wp_get_block_css_selector( $block_type, array( 'filter', 'duotone' ), true );
+		/*
+		 * If the experimental duotone support was set, that value is to be
+		 * treated as a selector and requires scoping.
+		 */
+		$experimental_duotone = _wp_array_get( $block_type->supports, array( 'color', '__experimentalDuotone' ), false );
```

Comment:
I'm not sure the exact scenarios that `$block_type->supports` may not exist, but I would feel more confident in the change if the conditional was re-added either here or where it was above.

```suggestion
		$experimental_duotone = false;
		if ( $block_type && property_exists( $block_type, 'supports' ) ) {
			$experimental_duotone = _wp_array_get( $block_type->supports, array( 'color', '__experimentalDuotone' ), false );
		}
```

Reply:
Agreed, since we don't (and can't) use `block_has_support()` here, here we have to include those checks again.

Reply:
@ajlende Why is this check needed? We now pass a `WP_Block_Type` object into the method. There is no need to check to see if the object is an object and it will always have a property of supports.

Reply:
@spacedmonkey I am mindful that this is being checked in [`block_has_support()`](https://developer.wordpress.org/reference/functions/block_has_support/) as well, so there's probably / hopefully a reason for it. Better safe than sorry.

Reply:
> Why is this check needed?

From what I said earlier:

> I'm not sure the exact scenarios that $block_type->supports may not exist

Echoing what @felixarntz said

> this is being checked in [block_has_support()](https://developer.wordpress.org/reference/functions/block_has_support/) as well, so there's probably / hopefully a reason for it. Better safe than sorry.

Reply:
The reason why this check was there before, the following could return null 
```
$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
```

Now `get_selector` is passed a WP_Block_Type object ALWAYS. That property will allow exist. This check is wasteful and unneeded.

Reply:
It looks like the constructor of `WP_Block` also uses `WP_Block_Type_Registry`'s `get_registered()`, so it looks like that could possibly still return `null` if the block name isn't registered when the block is created.

https://github.com/WordPress/wordpress-develop/blob/cdf3ac093de45c2c2d67fe0635f3d8a513a97895/src/wp-includes/class-wp-block.php#L124-L130

Reply:
Adding some type checking, just in case.

Reply:
Although, to be fair, in the `render` method, it only gets checked once in `$is_dynamic`.

https://github.com/WordPress/wordpress-develop/blob/cdf3ac093de45c2c2d67fe0635f3d8a513a97895/src/wp-includes/class-wp-block.php#L217

Everything else just expects it to be there, so there's probably a bunch that would break if it didn't exist.